### PR TITLE
Sync turntables and transfertables in multiplayer mode

### PR DESCRIPTION
**Pull request by Csantucci**
https://blueprints.launchpad.net/or/+spec/sync-turntables-in-multiplayer
http://www.elvastower.com/forums/index.php?/topic/28591-operational-turntable/page__view__findpost__p__287727

### CODE REVIEW

File: Source/Orts.Simulation/MultiPlayer/Message.cs [hunk 1]
```diff
@@ -3893,4 +3896,104 @@ public override string ToString()
...
\ No newline at end of file
+#region MSGMovingTbl
+
+    public class MSGMovingTbl : Message
+    {
+        public string user, newTrainName, carID, firstCarIDOld, firstCarIDNew;
+        public MovingTable.submessagecode subMessageCode;
+        public int movingTableIndex;
+        public bool clockwise;
+        public float yangle;
```
Comment (Sharpe49) [inline Source/Orts.Simulation/MultiPlayer/Message.cs:3907]: The name of public members should start with an upper case character.

File: Source/Orts.Simulation/MultiPlayer/Message.cs [hunk 2]
```diff
@@ -2779,8 +2781,9 @@ public override void HandleMsg()
             }
         }
     }
-#endregion MSGUncouple
+    #endregion MSGUncouple
```
Comment (Sharpe49) [inline Source/Orts.Simulation/MultiPlayer/Message.cs:2784]: Why is this being indented?

File: Source/Orts.Simulation/MultiPlayer/Message.cs [hunk 3]
```diff
@@ -2345,7 +2347,7 @@ public override void HandleMsg()
         }
 
     }
-#endregion MSGGetTrain
+    #endregion MSGGetTrain
```
Comment (Sharpe49) [inline Source/Orts.Simulation/MultiPlayer/Message.cs:2350]: Why is this being indented?

File: Source/Orts.Simulation/Simulation/Transfertables.cs [hunk 4]
```diff
@@ -172,15 +173,27 @@ protected void InitializeOffsetsAndTrackNodes()
...
+        }
+
+        public void GeneralComputeTarget(bool isForward)
         {
             if (!Continuous) return;
             Continuous = false;
             GoToTarget = false;
             Forward = isForward;
             Reverse = !isForward;
+            OffsetDiff = RemotelyControlled ? 2.8f : 1.4f;
```
Comment (Sharpe49) [inline Source/Orts.Simulation/Simulation/Transfertables.cs:194]: Maybe, put this formula in a getter (at line 50)

File: Source/Orts.Simulation/Simulation/Turntables.cs [hunk 5]
```diff
@@ -81,6 +81,11 @@ public class MovingTable
         protected int[] MyTrVectorSectionsIndex;
         public bool[] MyTrackNodesOrientation { get; protected set; } // true if forward, false if backward
         public int TrackShapeIndex;
+        public enum submessagecode
```
Comment (Sharpe49) [inline Source/Orts.Simulation/Simulation/Turntables.cs:84]: This should be in CamelCase.

**Comment by Csantucci** [comment]
Code cleaned up as requested

File: Source/Orts.Simulation/Simulation/Turntables.cs (hunk 5 above)
Comment (Sharpe49) [reply Source/Orts.Simulation/Simulation/Turntables.cs]: Since this an enum type definition, this should be UpperCamelCase, not lowerCamelCase.

Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Turntables.cs]: > Since this an enum type definition, this should be UpperCamelCase, not lowerCamelCase.

Done.

**Review by Sharpe49 — APPROVED** [review]

**Comment by Csantucci** [comment]
Thanks. Waiting for the approval of the blueprint before merging.

**Comment by cjakeman** [comment]
Hi Carlo,

> Thanks. Waiting for the approval of the blueprint before merging.

Blueprint approved, but the PR now has a conflict.

**Comment by Csantucci** [comment]
Thanks Chris. Conflict removed.
Thaks Cédric for immediate renewed approval.

**Comment by Csantucci** [comment]
Waiting a couple of days before merging.

**Pull request merged**